lib/Service/Avatar/SvgPortableSecureImage.php

```diff
@@ -0,0 +1,1173 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright 2020 Gregor Mitzka <gregor.mitzka@gmail.com>
+ *
+ * @author Gregor Mitzka <gregor.mitzka@gmail.com>
+ *
+ * Based on the work of Mouadd Boukiaou (2020).
+ *
+ * Mail
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Mail\Service\Avatar;
+
+use Closure;
+use Imagick;
+use ImagickPixel;
+use SimpleXMLElement;
+use Throwable;
+//use OCA\Mail\Exception\ServiceException;
+
+/**
+ * @see <a href="https://github.com/authindicators/svg-ps-converters/blob/master/illustrator-script/SaveAsSVGTinyPS_v0.1.jsx" />
+ * @see <a href="https://www.w3.org/TR/SVGTiny12/intro.html" />
+ * @see <a href="https://bimigroup.org/creating-bimi-svg-logo-files/" />
+ */
+class SvgPortableSecureImage {
+	/**
+	 * List of allowed element names.
+	 *
+	 * @var string[]
+	 */
+	private static $ALLOWED_ELEMENT_NAMES = [
+		'circle',
+		'defs',
+		'desc',
+		'ellipse',
+		'g',
+		'line',
+		'linearGradient',
+		'path',
+		'polygon',
+		'polyline',
+		'radialGradient',
+		'rect',
+		'solidColor',
+		'svg',
+		'text',
+		'textArea',
+		'title',
+		'use',
+	];
+
+	/**
+	 * List of allowed attributes for each allowed element type.
+	 *
+	 * @var array<string,string[]>
+	 */
+	private static $ALLOWED_ATTRIBUTES = [
```

Comment:
I know that this class is huge. Would it be better to put the static attributes into an associated JSON file and load it when the class is instantiated the first time?